src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -55,19 +57,22 @@ public abstract class PredictionTransformerBase<TModel, TScorer> : IPredictionTr
 
         protected abstract TScorer Scorer { get; set; }
 
-        protected PredictionTransformerBase(IHost host, TModel model, Schema trainSchema)
+        [BestFriend]
```

Comment:
>[BestFriend] [](start = 8, length = 12)

Is it necessary?

Reply:
Yes, there are a couple subclasses of this abstract class in other assemblies that call this attribute.

Indeed, as you work on #1995 for `IDataTransform` I'd expect you will need to lock down some of the abstract classes of dataview to make external subclassing impossible, which will require a similar approach, I expect?